database.py

```
@@ -0,0 +1,16 @@
+from sqlalchemy.ext.asyncio import AsyncSession, create_async_engine
+from sqlalchemy.orm import sessionmaker, DeclarativeBase
+from settings import settings
+
+SQLALCHEMY_DATABASE_URL = settings.DATABASE_URL
+
+engine = create_async_engine(
+    SQLALCHEMY_DATABASE_URL, connect_args={"check_same_thread": False}
+)
+SessionLocal = sessionmaker(
+    autocommit=False, autoflush=False, bind=engine, class_=AsyncSession
```

Session factory creation is correct for async usage, but consider explicit options like `expire_on_commit=False` if your code expects objects to remain accessible after commit. Also ensure any startup validation for DATABASE_URL occurs before session creation to fail fast.